weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx

```diff
@@ -0,0 +1,315 @@
+          setAnchorEl(null);
+          setSelectionState({allChecked: true});
+        }}>
+        <Tailwind>
+          <div className="min-w-[460px] p-12">
+            <div className="flex items-center pb-8">
+              <div className="flex-auto text-xl font-semibold">Export</div>
+              <div className="ml-16 text-moon-500">
+                {maybePluralize(visibleColumns.length, 'column', 's')}
+              </div>
+            </div>
+            <SelectionCheckboxes
+              selectedCalls={selectedCalls}
+              numTotalCalls={numTotalCalls}
+              selectionState={selectionState}
+              setSelectionState={setSelectionState}
+            />
+            <DownloadGrid onClickDownload={onClickDownload} />
+            <div className="mt-12 flex items-center text-moon-500">
+              <IconOnlyPill color="moon" icon="warning" />
+              <div className="ml-6">
+                This export experience is in beta and subject to change.
+              </div>
+            </div>
+          </div>
+        </Tailwind>
+      </Popover>
+    </>
+  );
+};
+
+const SelectionCheckboxes: FC<{
+  selectedCalls: string[];
+  numTotalCalls: number;
+  selectionState: SelectionState;
+  setSelectionState: Dispatch<SetStateAction<SelectionState>>;
+}> = ({selectedCalls, numTotalCalls, selectionState, setSelectionState}) => {
+  return (
+    <div className="flex items-center">
+      <div className="ml-2" />
+      <Checkbox
+        checked={selectionState.allChecked ?? false}
+        onCheckedChange={() => setSelectionState(s => ({allChecked: true}))}
+      />
+      <div className="ml-6 mr-24">all rows ({numTotalCalls})</div>
+      <Checkbox
+        checked={selectionState.selectedChecked ?? false}
+        onCheckedChange={() =>
+          setSelectionState(s => ({
+            selectedChecked: !s.selectedChecked,
+            allChecked: s.selectedChecked,
+          }))
+        }
+      />
+      <div className="ml-8 mr-6">first</div>
+      <div className="w-auto min-w-[50px]">
+        <TextField
+          type="number"
+          placeholder="1000"
+          value={selectionState.limit ?? ''}
+          onChange={value =>
+            setSelectionState(s => ({
+              selectedChecked: s.selectedChecked,
+              limit: value,
+            }))
+          }
+        />
+      </div>
+      <div className="ml-4 mr-16">rows</div>
+      {selectedCalls.length > 0 && selectedCalls.length !== numTotalCalls && (
+        <>
+          <Checkbox
+            checked={selectionState.limitChecked ?? false}
+            onCheckedChange={() =>
+              setSelectionState(s => ({
+                limitChecked: !s.limitChecked,
+                allChecked: s.limitChecked,
+              }))
+            }
+          />
+          <div className="ml-4">selected rows ({selectedCalls.length})</div>
+        </>
+      )}
+    </div>
+  );
+};
+
+const ClickableOutlinedCardWithIcon: FC<{
```

Comment:
I think this component feels weird without a hover interaction and with the repeated icon, maybe follow up with design?